sql/src/test/java/io/crate/execution/engine/window/WindowFunctionTest.java

```diff
@@ -0,0 +1,170 @@
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.engine.window;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.breaker.RamAccountingContext;
+import io.crate.expression.InputFactory;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.Functions;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.doc.DocSchemaInfo;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.TestingTableInfo;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.testing.SqlExpressions;
+import io.crate.testing.TestingBatchIterators;
+import io.crate.types.ArrayType;
+import io.crate.types.DataTypes;
+import io.crate.types.SetType;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.junit.Before;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+
+class WindowFunctionTest {
+
+    private SqlExpressions sqlExpressions;
+    private Functions functions;
+    private Map<QualifiedName, AnalyzedRelation> tableSources;
+    private TransactionContext txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+    private InputFactory inputFactory;
+
+    private final RamAccountingContext RAM_ACCOUNTING_CONTEXT =
+        new RamAccountingContext("dummy", new NoopCircuitBreaker("dummy"));
+
+    @Before
+    public void prepareFunctions() throws Exception {
+        DocTableInfo tableInfo = TestingTableInfo.builder(new RelationName(DocSchemaInfo.NAME, "users"), null)
+            .add("id", DataTypes.INTEGER)
```

Comment:
I'd strip this down to what's used/needed now and extend it as we go - instead of start with a blown up version. We might end-up not needing a lot of these columns.